stdlib/camlinternalFormat.ml

```diff
@@ -809,6 +822,9 @@ and trans : type
   | Int64_ty rest1, Int64_ty rest2 -> Int64_ty (trans rest1 rest2)
   | Nativeint_ty rest1, Nativeint_ty rest2 -> Nativeint_ty (trans rest1 rest2)
   | Float_ty rest1, Float_ty rest2 -> Float_ty (trans rest1 rest2)
+  | Binary_ty rest1, Binary_ty rest2 -> Binary_ty (trans rest1 rest2)
+  | String_ty _, Binary_ty _ -> assert false
+  | Binary_ty _, String_ty _ -> assert false
```

Comment:
Im not sure why these last two are needed but the type checker complains if they are not present - incomplete pattern match error?

Reply:
The issue is that it is incorrect to have a constructor `Binary_ty` in the no-safe-string configuration mode. Indeed, in this case `string`=`bytes` and the `Binary_ty` constructor should be merged with the `String_ty` constructor.

Reply:
ah ok, got it, thanks - at least I get the no-safe-string and ```bytes``` = ```string``` part.  

Not sure I get the merging idea yet though - I'll need to think about how to achieve that more - assuming we continue with this PR....

Reply:
so I think I misunderstood the first time around, 

> in this case string=bytes and the Binary_ty constructor should be merged with the String_ty constructor

do you mean that the merging should happen during compilation or that this PR should implement a merged version?